src/toil_scripts/adam_gatk_pipeline/align_and_call.py

```diff
@@ -0,0 +1,243 @@
+                        help = 'hapmap_3.3.b37.vcf URL')
+    
+    # return built parser
+    return parser
+
+def static_dag(job, s3_bucket, uuid, bwa_inputs, adam_inputs, gatk_inputs):
+    """
+    Prefer this here as it allows us to pull the job functions from other jobs
+    without rewrapping the job functions back together.
+
+    bwa_inputs: Input arguments to be passed to BWA.
+    adam_inputs: Input arguments to be passed to ADAM.
+    gatk_inputs: Input arguments to be passed to the GATK.
+    """
+
+    # get work directory
+    work_dir = job.fileStore.getLocalTempDir()
+
+    # does the work directory exist?
+    if not os.path.exists(work_dir):
+        os.mkdirs(work_dir)
+
+    # write config for bwa
+    bwafp = open("%s_bwa_config.csv" % uuid, "w")
+    bwa_inputs['config'] = "%s_bwa_config.csv" % uuid
+    print >> bwafp, "%s,https://s3.amazonaws.com/%s/sequence/%s_1.fastq.gz,https://s3.amazonaws.com/%s/sequence/%s_2.fastq.gz" % (uuid, s3_bucket, uuid, s3_bucket, uuid)
+    bwafp.flush()
+    bwafp.close()
+
+    # write config for gatk
+    gatkfp = open("%s_gatk_config.csv" % uuid, "w")
+    gatk_inputs['config'] = "%s_gatk_config.csv" % uuid
+    print >> gatkfp, "%s,https://s3.amazonaws.com/%s/analysis/%s.bam" % (uuid, s3_bucket, uuid)
+    gatkfp.flush()
+    gatkfp.close()
+    
+    # get head bwa job function and encapsulate it
+    bwa = job.wrapJobFn(toil_scripts.batch_alignment.bwa_alignment.download_shared_files,
+                        bwa_inputs).encapsulate()
+
+    # get head ADAM job function and encapsulate it
+    adam = job.wrapJobFn(toil_scripts.adam_pipeline.spark_toil_script.start_master,
+                         adam_inputs).encapsulate()
+    
+    # get head GATK job function and encapsulate it
+    gatk = job.wrapJobFn(toil_scripts.gatk_germline.germline.batch_start,
+                         gatk_inputs).encapsulate()
+
+    # wire up dag
+    job.addChild(bwa)
+    bwa.addChild(adam)
+    adam.addChild(gatk)
+
+if __name__ == '__main__':
+    
+    args_parser = build_parser()
+    Job.Runner.addToilOptions(args_parser)
+    args = args_parser.parse_args()
+
+    bwa_inputs = {'ref.fa': args.ref,
+                  'ref.fa.amb': args.amb,
+                  'ref.fa.ann': args.ann,
+                  'ref.fa.bwt': args.bwt,
+                  'ref.fa.pac': args.pac,
+                  'ref.fa.sa': args.sa,
+                  'ref.fa.fai': args.fai,
+                  'ref.fa.alt': args.alt,
+                  'ssec': None,
+                  'output_dir': None,
+                  'sudo': args.sudo,
+                  's3_dir': "%s/alignment" % args.s3_bucket,
+                  'lb': args.uuid,
+                  'uuid': args.uuid,
+                  'cpu_count': None,
+                  'use_bwakit': args.use_bwakit}
+    
+    adam_inputs = {'numWorkers': args.num_nodes - 1,
+                   'outDir':     's3://%s/analysis/%s.bam' % (args.s3_bucket, args.uuid),
```

Comment:
On further consideration, the suggestion was more for long term storage in s3 rather than working names during processing, which actually use uuids instead of sample_ids.